### ISSUE TITLE
Feature: New Frontend Design / SPA as Frontend

### ISSUE THREAD

**Issue by florianstahr** [CodeDesignInitiative/campy#86]
# New Frontend Design
I associate with a product made by Code+Design something unique or more an experience that includes a high quality design no matter the size of a product. So I made a design that I think could be a kick start to bring my associations to reality and that could be applied to the campy dashboard.

![landing_login_desktop-2_1](https://user-images.githubusercontent.com/29822454/41739285-e889a288-7594-11e8-97f4-c425e1861e86.jpg)
![landing_login_mobile-1_1](https://user-images.githubusercontent.com/29822454/41739286-e8a9813e-7594-11e8-86e1-a6d6e56339dc.jpg)

# SPA as Frontend
I think it would be much easier to implement that design within a SPA rather than within a MPA. I know this means a totally new structure of the project and because of that I would like to start building a SPA with React as a side project that could be merged if it work at some point.

**Comment by Jubeki** [comment]
I think the new design is a good idea.
To my mind, the font-family should be more smooth and be not so edged.

Most people don't know how to work with a Single Page Application.
More people know PHP or Blade.
So I don't think this is applicable.

But generally nice idea.

**Comment by minthemiddle** [comment]
You actually address two completely different topics, @florianstahr:

1. *Design: * Yes, Campy does not look pretty. And it has UX flaws. PR welcome.
2. *SPA:* Has no priority at the moment, you should go for a fork right now (reasoning below)

Switching from a PHP backend/HTML+CSS frontend to a SPA with PHP backend and JS frontend will not happen anytime soon in the main project:

- This would require a complete rewrite to API driven
- JS would introduce more tools (building, testing)
- We would need to test both frontend and backend with different tools
- The main maintainers (me, @Jubeki, @toorusr) have very little experience with JS, especially testing
- The backend needs way more polishing (complete test coverage, refactoring) and introducing a new technology and more complexity would be a big distraction to achieving these goals
- The app consists of only a handful of views that load super quick, introducing JS would be overkill in my opinion

TLDR: PR and forks very welcome, design will improve, SPA will not be a priority in the main project in the foreseeable future!
